lua/xcodebuild/platform/macos.lua

```diff
@@ -14,7 +14,9 @@ local M = {}
 ---@param callback function|nil
 ---@return number # job id
 function M.launch_app(appPath, callback)
-  local command = { "open", appPath }
+  local appName = util.get_filename(appPath)
+  local executablePath = appPath .. "/Contents/MacOS/" .. appName
+  local command = { executablePath }
 
   local runArgs = appdata.read_run_args()
```

Comment:
Unfortunately, passing args will stop working now, it was an option for `open` command which is no longer use.
I believe you could just remove `--args`, but please verify if it works.